TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/RoutesToProfessionalStatus/EditRoute/Status.cshtml.cs

```diff
@@ -0,0 +1,117 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.DataStore.Postgres.Models;
+using TeachingRecordSystem.SupportUi.Infrastructure.Filters;
+
+namespace TeachingRecordSystem.SupportUi.Pages.RoutesToProfessionalStatus.EditRoute;
+
+[Journey(JourneyNames.EditRouteToProfessionalStatus), RequireJourneyInstance, CheckProfessionalStatusExistsFilterFactory()]
+public class StatusModel(
+    TrsLinkGenerator linkGenerator,
+    ReferenceDataCache referenceDataCache) : PageModel
+{
+    public string? PersonName { get; set; }
+    public Guid PersonId { get; private set; }
+    public JourneyInstance<EditRouteState>? JourneyInstance { get; set; }
+    public RouteToProfessionalStatus Route { get; set; } = null!;
+
+    public ProfessionalStatusStatusInfo[] Statuses { get; set; } = [];
+
+    [FromQuery]
+    public bool FromCheckAnswers { get; set; }
+
+    [FromRoute]
+    public Guid QualificationId { get; set; }
+
+    [BindProperty]
+    [Display(Name = "Select the route status")]
+    [Required(ErrorMessage = "Select a route status")]
+    public ProfessionalStatusStatus Status { get; set; }
+
+    public void OnGet()
+    {
+        Status = JourneyInstance!.State.Status;
+    }
+
+    public async Task<IActionResult> OnPostAsync()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        if (CompletingRoute)
+        {
+            var hasImplicitExemption =
+                Route.InductionExemptionReasonId.HasValue ?
+                    (await referenceDataCache.GetInductionExemptionReasonByIdAsync(Route.InductionExemptionReasonId.Value)).RouteImplicitExemption
+                    : false;
+            await JourneyInstance!.UpdateStateAsync(
+                s =>
+                {
+                    s.EditStatusState = new EditRouteStatusState
+                    {
+                        Status = Status,
+                        RouteImplicitExemption = hasImplicitExemption,
+                        InductionExemption = hasImplicitExemption ? true : null
+                    };
+                });
+        }
+        else if (NotCompletedRoute)
+        {
+            await JourneyInstance!.UpdateStateAsync(s =>
+            {
+                s.AwardedDate = null;
+                s.IsExemptFromInduction = null;
+                s.Status = Status;
+            });
+        }
+        else // going from Approved to Awarded or vice versa
```

Comment:
should this be able to happen? worth us finding out even if we just merge this code for now

Reply:
question opened up to team on Slack